packages/protocol/src/action-protocol/clipboard.ts

```diff
@@ -14,80 +14,154 @@
+export namespace SetClipboardDataAction {
+    export const KIND = 'setClipboardData';
+
+    export function is(object: any): object is SetClipboardDataAction {
+        return Action.hasKind(object, KIND) && hasObjectProp(object, 'clipboardData');
+    }
+
+    export function create(clipboardData: ClipboardData, options: { responseId?: string } = {}): SetClipboardDataAction {
+        return {
+            kind: KIND,
+            responseId: '',
+            clipboardData,
+            ...options
+        };
+    }
 }
 
 /**
  * Requests a cut operation from the server, i.e., deleting the selected elements from the model. Before submitting a `CutOperation`
  * a client should ensure that the cut elements are put into the clipboard.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `CutOperations`.
  */
-export class CutOperation implements Operation {
-    static readonly KIND = 'cut';
+export interface CutOperation extends Operation {
+    /**
+     * The unique action kind.
+     */
+    kind: typeof CutOperation.KIND;
 
-    constructor(public readonly editorContext: EditorContext, public readonly kind: string = CutOperation.KIND) {}
+    /**
+     * The current editor context.
+     */
+    editorContext: EditorContext;
 }
 
-export function isCutOperation(action: any): action is CutOperation {
-    return isActionKind(action, CutOperation.KIND) && isObject(action, 'editorContext');
+export namespace CutOperation {
+    export const KIND = 'cut';
+
+    export function is(object: any): object is CutOperation {
+        return Operation.hasKind(object, KIND) && hasObjectProp(object, 'editorContext');
+    }
+
+    export function create(editorContext: EditorContext): CutOperation {
+        return {
+            kind: KIND,
+            isOperation: true,
+            editorContext
+        };
+    }
 }
 
 /**
  * Requests a paste operation from the server by providing the current clipboard data. Typically this means that elements should be created
  *  based on the data in the clipboard.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `PasteOperations`.
  */
-export class PasteOperation implements Operation {
-    static readonly KIND = 'paste';
-
-    constructor(
-        public readonly clipboardData: ClipboardData,
-        public readonly editorContext: EditorContext,
-        public readonly kind: string = PasteOperation.KIND
-    ) {}
-}
+export interface PasteOperation extends Operation {
+    /**
+     * The unique action kind.
+     */
+    kind: typeof PasteOperation.KIND;
 
-export function isPasteOperation(action: any): action is PasteOperation {
-    return isActionKind(action, PasteOperation.KIND) && isObject(action, 'clipboardData') && isObject(action, 'editorContext');
+    /**
+     * The current editor context.
+     */
```

Comment:
I'd spare such comments, also if it hurts to be "incomplete". But imho they just add noise and add no information.